### PR TITLE
Remove "run on work queue" bits from non-concrete controllers.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
There is no "work queue" in the sense of the Matter queue involved unless we have a concrete controller.

* syncRunOnWorkQueueWithReturnValue was unused on MTRDeviceController, so can just be removed.

* syncRunOnWorkQueueWithBoolReturnValue was unused on MTRDeviceController, so can just be removed.

* After that syncRunOnWorkQueue becomes unused and can be removed.

* At this point chipWorkQueue is not used on the base class and can be moved into the subclasses as needed.

### CODE REVIEW

File: src/darwin/Framework/CHIP/MTRDeviceController_Internal.h [hunk 1]
```diff
@@ -71,7 +71,6 @@ NS_ASSUME_NONNULL_BEGIN
 
 // queue used to serialize all work performed by the MTRDeviceController
 // (moved here so subclasses can initialize differently)
```
Comment (vivien-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController_Internal.h:73]: Do we need to keep the comment ?

File: src/darwin/Framework/CHIP/MTRDeviceController_Internal.h [hunk 2]
```diff
@@ -71,7 +71,6 @@ NS_ASSUME_NONNULL_BEGIN
 
 // queue used to serialize all work performed by the MTRDeviceController
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController_Internal.h:72]: The comments for the removed property could be removed as well.

**Review by vivien-apple — APPROVED** [review]

**Review by jtung-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDeviceController_Internal.h (hunk 2 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTRDeviceController_Internal.h:72]: Good catch, removing.

Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController_Internal.h:72]:
```suggestion
```

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform-specific change with platform owner review.

**Pull request merged**